src/components/posts/video-player-overlay.tsx

```diff
@@ -0,0 +1,144 @@
+import Link from 'next/link'
+import Image from 'next/image'
+
+import {
+  useVideoPlayerOverlay,
+  type CompletedAction,
+} from '@/hooks/mux/use-video-player-overlay'
+import {Post} from '@/pages/[post]'
+import Spinner from '@/spinner'
+import {Button} from '@/ui/button'
+import {ArrowRight} from 'lucide-react'
+import {RefreshCw} from 'lucide-react'
+import {Card, CardContent, CardTitle, CardHeader, CardFooter} from '@/ui/card'
+import {motion} from 'framer-motion'
+
+type VideoPlayerOverlayProps = {
+  resource: Post
+}
+
+const VideoPlayerOverlay: React.FC<VideoPlayerOverlayProps> = ({resource}) => {
+  const {state: overlayState, dispatch} = useVideoPlayerOverlay()
+
+  switch (overlayState.action?.type) {
+    case 'COMPLETED':
+      const {playerRef, cta} = overlayState.action as CompletedAction
+
+      if (cta === 'cursor_workshop') {
```

Comment:
if cta is `cursor_workshop` display the custom overlay, otherwise display default